src/cargo/util/paths.rs

```diff
@@ -547,3 +547,36 @@ fn exclude_from_time_machine(path: &Path) {
     // Errors are ignored, since it's an optional feature and failure
     // doesn't prevent Cargo from working
 }
+
+/// Absolute path to an on-disk temporary directory.
+/// It may be system-wide or user-specific depending on system conventions.
+/// Used in layout.rs, only needed on Unix systems.
+#[cfg(unix)]
+pub fn persistent_temp_path() -> Option<PathBuf> {
+    #[cfg(target_os = "macos")]
+    {
+        // "Library/Caches" should be obtained via NSFileManager's
+        // URLsForDirectory:NSCachesDirectory inDomains:NSUserDomainMask
+        // However, cocoa-foundation doesn't have bindings for it yet.
+        // This path has remained stable for two decades,
+        // so hardcode it for simplicity (dirs crate does the same).
+        Some(home::home_dir()?.join("Library/Caches/Cargo"))
+    }
+    #[cfg(not(target_os = "macos"))]
+    {
+        // XDG standard
+        if let Some(path) = env::var_os("XDG_CACHE_HOME") {
+            let path = PathBuf::from(path);
+            if path.is_absolute() {
+                return Some(path.join("cargo"));
+            }
+        }
+        // FHS standard
+        // This is not using /tmp, because /tmp could be using ramfs.
+        let path = Path::new("/var/tmp");
+        if path.exists() {
+            return Some(path.join("cargo"));
+        }
+        None
```

Comment:
Not sure I understand this logic – is there any reason why it ignores the "important" part of XDG and instead does something completely altogether in the "env not set" case?

Reply:
What "important" part do you mean?

I am afraid that touching XDG is going to attract bikeshed fiesta. I'm tempted to remove it for that reason, and go straight for `/var/tmp` instead.

Reply:
> What "important" part do you mean?

XDG defines default directories if the env var is not set. The env var is probably not set in 99% of the cases, because people are fine with the defaults.

> I am afraid that touching XDG is going to attract bikeshed fiesta.

Did you consider simply using a library?

> I'm tempted to remove it for that reason, and go straight for /var/tmp instead.

That's a cheap way out for this special case, but I'm not sure you want the next person following this precedent when they try to fix the issue for config/data files.

Reply:
I've considered using a library, but it this area is full of abandoned projects and half-abandoned forks, so I didn't see any suitable crate.

I think the best option would be to use a cargo-specific variable/setting for customizing this dir.

Reply:
> I've considered using a library, but it this area is full of abandoned projects and half-abandoned forks, so I didn't see any suitable crate.

I think no crate can help you if your intention is to invent your own non-standard scheme anyway.

> I think the best option would be to use a cargo-specific variable/setting for customizing this dir.

That's of course technically possible, but is does nothing to resolve https://github.com/rust-lang/rfcs/pull/1615 which is about sane _defaults_.

Reply:
> I think no crate can help you if your intention is to invent your own non-standard scheme anyway.

Jabs like that are unhelpful. Please don't do that.

Reply:
Have a nice day.